tools/generate_release_notes.py

```diff
@@ -79,125 +114,190 @@ def tqdm(i, **kwargs):
         previous_tag = tag
         break
 else:
-    raise RuntimeError(f"Desired tag ({args.from_commit}) not found")
+    raise RuntimeError(f'Desired tag ({args.from_commit}) not found')
+
+common_ancestor = local_repo.merge_base(args.to_commit, args.from_commit)[0]
+remote_commit = repository.get_commit(common_ancestor.hexsha)
 
 # For some reason, go get the github commit from the commit to get
 # the correct date
-github_commit = previous_tag.commit.commit
 previous_tag_date = datetime.strptime(
-    github_commit.last_modified, "%a, %d %b %Y %H:%M:%S %Z"
+    remote_commit.last_modified, '%a, %d %b %Y %H:%M:%S %Z'
 )
 
 
-all_commits = list(
+def get_commits_to_ancestor(ancestor, rev="main"):
+    yield from local_repo.iter_commits(f'{ancestor.hexsha}..{rev}')
+
+
+new_commits_count = (
+    len(list(get_commits_to_ancestor(common_ancestor, args.to_commit))) + 1
+)
+release_branch_count = (
+    len(list(get_commits_to_ancestor(common_ancestor, args.from_commit))) + 1
+)
+
+with short_cache(60):
+    all_commits = list(
+        tqdm(
+            repository.get_commits(
+                sha=args.to_commit, since=previous_tag_date
+            ),
+            desc=f'Getting all commits between {remote_commit.sha} '
+            f'and {args.to_commit}',
+            total=new_commits_count,
+        )
+    )
```

Comment:
So I may be getting this completely wrong but it seems like the caching may not be working? First of all the context is only covering getting the commits. Everything else happens after the context manager `short_cache` is closed. Secondly, even if indent everything after this a second execution seems to take the exact time as before. It seems like there is no caching happening?

@Czaki, I'm not sure whether you are the right person to ping on this but maybe you would know more? :pray:

Reply:
Caching is not introduced for speedup but to reduce the probability of hitting the GitHub limit of API access (5000 per hour if I remember correctly) 

But it caches every GitHub API call. Even a list of all commits since the given tag. So I created a context manager that shorted the cache for this operation to 1 minute. 

But As I remember, the speedup could be observed if you comment out lines 52-54 that setup cache.

Reply:
Ah, good to know! I just discovered this by accident while playing around with it. It's nice to have that option while iterating on the script. Thanks for the fast reply. :blush: